core/src/memzero.c

```diff
@@ -1,7 +1,23 @@
-#include <stdint.h>
-
 #include "memzero.h"
 
+#ifdef CONFIG_HAVE_MEMSET_S
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include "log.h"
+
+void memzero(void* s, size_t n) {
+  int res = memset_s(s, n, 0, n);
+  if (res != 0) {
+    ERROR("memset_s() failed with error code %d: %s", (int) res, strerror(res));
+    abort(); // unsafe to continue, terminate the program
+  }
```

Comment:
note: this will only fail if s == NULL or n > RSIZE_MAX according to the docs. So it shouldn't fail in practice.